sky/resources.py

```diff
@@ -632,10 +632,12 @@ def _try_validate_image_id(self) -> None:
                     'Cloud must be specified when image_id is provided.')
 
         if not self._cloud.is_same_cloud(
-                clouds.AWS()) and not self._cloud.is_same_cloud(clouds.GCP()):
+                clouds.IBM()) or self._cloud.is_same_cloud(
```

Comment:
```suggestion
                clouds.IBM()) and not self._cloud.is_same_cloud(
```

Reply:
Why are we using `or` previously?

Reply:
you're right. I got the wrong idea about what this segment supposed to check.